userspace/falco/configuration.cpp

```diff
@@ -176,8 +182,36 @@ void falco_configuration::load_engine_config(const std::string& config_name, con
 	}
 }
 
+
+void validateLoadedYAML(const YAML::Node& loadedYaml, const valijson::adapters::YamlCppAdapter& schemaAdapter) {
+	valijson::Validator validator;
+	validator.validate(schemaAdapter, loadedYaml);
+	if (!validator.isValid()) {
+		throw std::logic_error("ERROR: Loaded YAML does not conform to the schema");
+	}
+}
+
 void falco_configuration::load_yaml(const std::string& config_name, const yaml_helper& config)
 {
+	valijson::Schema schema;
+	valijson::SchemaParser parser;
+	std::ifstream schemaFile("falco_keys.json");
```

Comment:
if possible I would avoid using a file, I would love to use a string in this way it is all self-contained in the Falco binary. if we use the file we need to ship with Falco